pkg/server/config.go

```diff
@@ -42,6 +61,36 @@ func (cfg *HandlerConfig) Validate() error {
 	return nil
 }
 
+// ValidateMappingConfig validates the handler config for mapping service after load.
+func (cfg *HandlerConfig) ValidateMappingConfig() error {
+	if err := cfg.Validate(); err != nil {
+		return err
+	}
+
+	// For mapping server, we also require a failure topic ID.
+	if cfg.FailureTopicID == "" {
+		return fmt.Errorf("PMAP_FAILURE_TOPIC_ID is empty and require a value for mapping service")
+	}
+
+	if cfg.MappingConfig.DefaultResourceScope == "" {
+		return fmt.Errorf(`PMAP_RESOURCE_SCOPE is empty and require a value from one of the following format:\n
+			projects/{PROJECT_ID}\n
+			projects/{PROJECT_NUMBER}\n
+			folders/{FOLDER_NUMBER}\n
+			organizations/{ORGANIZATION_NUMBER}`)
+	}
+
+	scope := strings.Split(cfg.MappingConfig.DefaultResourceScope, "/")[0]
+	if _, ok := SupportedResourceScope[scope]; !ok {
+		return fmt.Errorf(`PMAP_RESOURCE_SCOPE: %s doesn't have a valid value, the ResourceScope should be one of the following formats:\n
+		projects/{PROJECT_ID}\n
+		projects/{PROJECT_NUMBER}\n
+		folders/{FOLDER_NUMBER}\n
+		organizations/{ORGANIZATION_NUMBER}`, cfg.MappingConfig.DefaultResourceScope)
+	}
```

Comment:
No need to use a map
```suggestion
  switch scope {
  case "projects", "folders", "organizations":
    // continue
  default:
    return fmt.Errorf("PMAP_MAPPING_RESOURCE_SCOPE is required in one of the formats: %v", allowedScopes)
  }
```

Reply:
It will be used here too https://github.com/abcxyz/pmap/blob/main/pkg/mapping/processors/assetinventory.go#L250, might  be better to make it a var.

Reply:
I would also think using a map is better.

Reply:
No, the processor package (where asset inventory processor lives) should not take a dependency on the server package (where the config lives). They should be decoupled.